sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -620,7 +667,9 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
 
     val orderedPartitionSpec = new util.LinkedHashMap[String, String]()
     getTable(db, table).partitionColumnNames.foreach { colName =>
-      orderedPartitionSpec.put(colName, partition(colName))
+      // Lowercase the partition column names before passing the partition spec to Hive client, as
+      // Hive metastore is not case preserving.
```

Comment:
Let's mention that Hive will lower case the part column names. So, if we do not convert the name to its lower case, hive will complain.